product_state/models/product_template.py

```diff
@@ -1,18 +1,72 @@
 # Copyright 2017 ACSONE SA/NV (<http://acsone.eu>)
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
 
-from odoo import models, fields
+from odoo import api, models, fields
+
+
+class ProductState(models.Model):
+    _name = 'product.state'
+    _description = "Product State"
+    _order = 'sequence, id'
+
+    name = fields.Char('State Name', required=True, translate=True)
+    code = fields.Char('State Code', required=True)
+    sequence = fields.Integer('Sequence', help="Used to order the States", default=25)
+    description = fields.Text(translate=True)
+    product_ids = fields.One2many(
+        'product.template',
+        'product_state_id',
+        string='State Products',
+    )
+    products_count = fields.Integer(
+        string='Number of products',
+        compute='_compute_products_count',
+    )
+
+    _sql_constraints = [
+        ('product_state_code_unique', 'UNIQUE(code)', 'Product State Code must be unique.')
```

Comment:
product_state/models/product_template.py:27:89: E501 line too long (91 > 88 characters)
write code_unique instead of product_sate_code_unique